src/foam/nanos/ticket/Ticket.js

```diff
@@ -325,7 +337,62 @@ foam.CLASS({
       class: 'Reference',
       of: 'foam.nanos.auth.User',
       name: 'assignedTo',
-      section: 'infoSection'
+      section: 'infoSection',
+      postSet: function(_, n) {
+        if ( n != 0 ) {
+          this.assignedToGroup = '';
+        }
+      },
+      order: 7,
+      gridColumns: 6
+    },
+    {
+      class: 'Reference',
+      of: 'foam.nanos.auth.Group',
+      name: 'assignedToGroup',
+      section: 'infoSection',
+      postSet: function(_, n) {
+        if ( n !== '' ) {
+          this.assignedTo = 0;
+        }
+      },
+      order: 8,
+      gridColumns: 6
+    },
+    {
+      class: 'Reference',
+      of: 'foam.nanos.auth.User',
+      name: 'createdFor',
+      documentation: 'User/business this ticket was created for.',
+      section: 'infoSection',
+      readVisibility: 'RO',
+      updateVisibility: 'RO',
+      order: 6,
+      gridColumns: 6
+    },
+    {
+      class: 'String',
+      name: 'externalComment',
+      storageTransient: true,
+      section: 'infoSection',
+      readVisibility: 'HIDDEN',
+      validationPredicates: [
+        {
+          args: ['id', 'title', 'comment', 'externalComment'],
+          predicateFactory: function(e) {
+            return e.OR(
+              e.AND(
+                e.EQ(foam.nanos.ticket.Ticket.ID, 0),
+                e.NEQ(foam.nanos.ticket.Ticket.TITLE, "")
+              ),
+              e.NEQ(foam.nanos.ticket.Ticket.COMMENT, ""),
+              e.NEQ(foam.nanos.ticket.Ticket.EXTERNAL_COMMENT, "")
+            );
+          },
+          errorString: 'Please provide a comment or external comment.'
```

Comment:
```suggestion
          errorString: 'Please provide a comment visible to the initiator of the ticket.'
```
or something to this effect.